influence_simulator/influence_simulator.py

```diff
@@ -0,0 +1,139 @@
+from csv import writer as csv_writer
+from typing import Callable, NamedTuple
+
+import networkx as nx
+from lightning_diffusion.models import IndependentCascadeModel
+from tqdm.auto import tqdm
+from tqdm.contrib.concurrent import process_map
+
+
+class Peak(NamedTuple):
+    infected: int
+    iteration: int
+
+
+class SimulationResult(NamedTuple):
+    node: int
+    simulation_length: int
+    exposed: int
+    not_exposed: int
+    peak_infected: int
+    peak_iteration: int
+
+
+class InfluenceSimulator:
+    def __init__(
+        self,
+        graph: nx.Graph | str,
+        infection_probability: float,
+        *,
+        random_state: int = None,
+    ) -> None:
+        if isinstance(graph, nx.Graph):
+            self.graph = graph
+        else:
+            self.graph = self.load_graph(graph)
+
+        self.infection_probability = infection_probability
+        self.random_state = random_state
+
+        self.result: list[SimulationResult] = []
+
+    def init_simulation(self, node: int) -> IndependentCascadeModel:
+        model = IndependentCascadeModel(
+            graph=self.graph,
+            initially_infected=0,
+            infection_probability=self.infection_probability,
+            random_state=self.random_state,
+        )
+
+        model.state[node] = "infected"
```

Comment:
Directly assigning states will make more complex when we would be using diffusion models with different set of the states. It may require some changes in lightning diffusion package and using here base class (DiffusionModel)

Reply:
Probably we should have some class which will be like 'base model states' then for specific model we can inherit and modify it or even moving some of current simulator functions directly to model like init_simulation, simulate_node

Reply:
Another approach to consider might be to write a simulator for each new diffusion model, which will be easier, but in that case I would put this file in some simulator module

Reply:
Good idea - I've applied those changes